### PR TITLE
Feature api and bloc for allcoinpage and coinbycategory

### PR DESCRIPTION
**Pull request by haitran0508**

### CODE REVIEW

File: lib/src/blocs/shortened_category/shortened_category_event.dart [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  @override
+  List<Object?> get props => [];
+}
+
+class ShortenedCategoryRequested extends ShortenedCategoryEvent {
+  ShortenedCategoryRequested();
+
+  @override
+  List<Object?> get props => [];
+}
```
Comment (HaiNguyenPV12) [inline lib/src/blocs/shortened_category/shortened_category_event.dart:13]: same as above

File: lib/src/screens/home_screen/pages/all_coin_page.dart [hunk 2]
```diff
@@ -1,201 +1,183 @@
...
+                                        showSearch(
+                                          context: context,
+                                          delegate: CoinsSearch(
+                                            coinsBloc:
+                                                context.read<CoinsListBloc>(),
+                                          ),
+                                        );
+                                      },
+                                      child: const SizedBox(
+                                        child: Text('Click to search...'),
```
Comment (HaiNguyenPV12) [inline lib/src/screens/home_screen/pages/all_coin_page.dart:125]: bring to constants

**Review by HaiNguyenPV12 — DISMISSED** [review]

File: lib/src/blocs/shortened_category/shortened_category_event.dart (hunk 1 above)
Comment (haitran0508) [reply lib/src/blocs/shortened_category/shortened_category_event.dart:13]: fixed

File: lib/src/blocs/coins_list/coins_list_event.dart [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  @override
+  List<Object?> get props => [];
+}
+
+class CoinsListRequested extends CoinsListEvent {
+  CoinsListRequested();
+
+  @override
+  List<Object?> get props => [];
+}
```
[the remark this answers is not in the document]
Comment (haitran0508) [reply lib/src/blocs/coins_list/coins_list_event.dart:13]: fixed

File: lib/src/screens/home_screen/pages/all_coin_page.dart (hunk 2 above)
Comment (haitran0508) [reply lib/src/screens/home_screen/pages/all_coin_page.dart]: fixed

**Review by thanhphamtrung — APPROVED** [review]

**Pull request merged**